react-backend/client/src/components/visitors/input.js

```diff
@@ -25,6 +25,7 @@ export default class Input extends Component {
           onChange={this.handleUserInput}
           value={this.state.userInput}
           className="Form__Input"
+          {...this.props}
```

Comment:
A safer pattern is to separate the props that you know you're going to use manually from rest, which you just want to pass through. You can use object rest spreading to do this easily.

```js
const { label, question, name, option, ...props } = this.props;

<input id={label} name={name} {...props} />
```
This prevents you from overwriting props that you've set manually, like would probably happen in this case, with `name` being set to `undefined` if `props.name` is empty, even if `props.option` is defined.

This happens because in the child component react does a shallow merge of the props
```
{ id: label, name: name, ...props }
```
And therefore any manually specified prop with the a key that exists in `props` would be overwritten.

Reply:
Alrighty, I think I've done it